pep-0458.txt

```diff
@@ -533,237 +778,241 @@ same metadata or distributions.  There are also issues with consistency on PyPI
-snapshot should safely coexist with any other snapshot, and be able to be
-deleted independently, without affecting any other snapshot.
+To keep TUF metadata on PyPI consistent with the highly volatile target files,
+consistent snapshots SHOULD be used. Each consistent snapshot captures the
+state of all known projects at a given time and MAY safely coexist with any
+other snapshot, or be deleted independently, without affecting any other
+snapshot.
 
-The solution presented in this PEP is that every metadata or data file managed
-by PyPI and written to disk MUST include in its filename the `cryptographic
-hash`__ of the file.  How would this help clients that use the TUF protocol to
-securely and consistently install or update a project from PyPI?
+To maintain consistent snapshots, all TUF metadata MUST, when written to disk,
+include a version number in their filename:
 
-__ https://en.wikipedia.org/wiki/Cryptographic_hash_function
+  VERSION_NUMBER.ROLENAME.json,
+    where VERSION_NUMBER is an incrementing integer, and ROLENAME is one of the
+    top-level metadata roles -- *root*, *snapshot* or *targets* -- or one of
+    the delegated targets roles -- *bins* or *bin-n*.
 
-The first step in the TUF protocol requires the client to download the latest
-*timestamp* metadata.  However, the client would not know in advance the hash
-of the *timestamp* associated with the latest snapshot.  Therefore, PyPI MUST
-redirect all HTTP GET requests for *timestamp* to the *timestamp* referenced in
-the latest snapshot.  The *timestamp* role is the root of a tree of
-cryptographic hashes that points to every other metadata that is meant to exist
-together (i.e., clients request metadata in timestamp -> snapshot -> root ->
-targets order).  Clients are able to retrieve any file from this snapshot
-by deterministically including, in the request for the file, the hash of the
-file in the filename.  Assuming infinite disk space and no `hash collisions`__,
-a client may safely read from one snapshot while PyPI produces another
+The only exception is the *timestamp* metadata file, whose version would not be known
+in advance when a client performs an update. The *timestamp* metadata
+lists the
+version of the *snapshot* metadata, which in turn lists the versions of the
+*targets* and delegated targets metadata, all as part of a given consistent
 snapshot.
 
+The *targets* or delegated targets metadata refer to the actual target
+files, including all of their cryptographic hashes as specified above.
+Thus, to mark a target file as part of a consistent snapshot it MUST, when
+written to disk, include its hash in its filename:
+
+  HASH.FILENAME
+    where HASH is the `hex digest`__ of the hash of the file contents and
+    FILENAME is the original filename.
+
+This means that there are multiple copies of every target file, one for each
+of the cryptographic hash functions specified above.
```

Comment:
If I understand correctly, this means that the size of PyPI's "on disk" size for target files must grow 3x: once for the existing files (to maintain backwards compatibility), once for the SHA2-256 hash and once for the SHA2-512 hash. Correct? If so I think this should probably be noted in the "What Additional Repository Files are Required on PyPI?" section above.